cpp/src/join/join_common_utils.cuh

```diff
@@ -322,6 +331,32 @@ __device__ void flush_output_cache(unsigned int const activemask,
   }
 }
 
+template <int num_warps, cudf::size_type output_cache_size>
+__device__ void flush_left_cache(unsigned int const activemask,
+                                 cudf::size_type const max_size,
+                                 int const warp_id,
+                                 int const lane_id,
+                                 cudf::size_type* current_idx,
+                                 cudf::size_type current_idx_shared[num_warps],
+                                 size_type join_shared_l[num_warps][output_cache_size],
+                                 size_type* join_output_l)
+{
+  int num_threads               = __popc(activemask);
```

Comment:
```suggestion
  int const num_threads               = __popc(activemask);
```

Reply:
My apologies, I'm a scala programmer and everything is `const` by default so I keep forgetting I have to be explicit in C++ with it 😂 it would be nice if `const` was the default actually.